website/addons/evernote/serializer.py

```diff
@@ -0,0 +1,77 @@
+from website.addons.base.serializer import OAuthAddonSerializer
+from website.util import api_url_for, web_url_for
+
+class EvernoteSerializer(OAuthAddonSerializer):
+
+    @property
+    def addon_serialized_urls(self):
+        node = self.node_settings.owner
+
+        return {
+            'accounts': node.api_url_for('evernote_get_user_settings'),
+        }
+
+
+    def serialize_settings(self, node_settings, current_user, client=None):
+        """View helper that returns a dictionary representation of a
+        EvernoteNodeSettings record. Provides the return value for the
+        evernote config endpoints.
+        """
+        valid_credentials = True
+        user_settings = node_settings.user_settings
+        self.node_settings = node_settings
+        current_user_settings = current_user.get_addon('evernote')
+        user_is_owner = user_settings is not None and user_settings.owner == current_user
+
+        # if user_settings:
```

Comment:
For the Box addon, these lines perform a simple (and arbitrary) API call to see if the credentials are valid. I'm not sure this is totally necessary, but might be simple to implement.